controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -969,6 +969,51 @@ func TestDeleteNSTemplateSet(t *testing.T) {
 		AssertThatNSTemplateSet(t, namespaceName, username, r.Client).
 			DoesNotHaveFinalizer() // finalizer was not added and nothing else was done
 	})
+
+	t.Run("NSTemplateSet not deleted until namespace is deleted", func(t *testing.T) {
+		// given an NSTemplateSet resource and 1 active user namespaces ("dev")
+		nsTmplSet := newNSTmplSet(namespaceName, username, "advanced", withNamespaces("abcde11", "dev"), withDeletionTs(), withClusterResources("abcde11"))
+		devNS := newNamespace("advanced", username, "dev", withTemplateRefUsingRevision("abcde11"))
+
+		r, fakeClient := prepareController(t, nsTmplSet, devNS)
+		req := newReconcileRequest(namespaceName, username)
+
+		fakeClient.MockDelete = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
+			fmt.Printf("Skipping deletion")
```

Comment:
Zombie?
```suggestion
```

Reply:
@alexeykazakov wrote tests here - https://github.com/codeready-toolchain/toolchain-e2e/pull/322 - but the desired scenario is not achieved. namespace is still left in terminating state. Checked the logs - it stops returning [this](https://github.com/ranakan19/member-operator/blob/CRT897/controllers/nstemplateset/namespaces.go#L188) after returning this error twice. I'm investigating the reason.

Reply:
Please delete this Printf